### PR TITLE
edi_storage: path configurable by type and param

### PR DESCRIPTION
**Pull request by simahawk**
TODO

- [x] add tests

### CODE REVIEW

**Comment by etobella** [comment]
TEsts fails. Solved on #584

File: edi_storage_oca/tests/test_components_base.py [hunk 1]
```diff
@@ -18,14 +18,14 @@ def test_remote_file_path(self):
             (("output", "error", "foo.csv"), "demo_out/error/foo.csv"),
         )
         for _args, expected in to_test:
-            path_obj = self.checker._remote_file_path(*_args)
+            direction, state, filename = _args
+            self.checker.exchange_record.direction = direction
```
Comment (simahawk) [inline edi_storage_oca/tests/test_components_base.py]: @jcoux same problem regarding overriding direction here?

Comment (jcoux) [reply edi_storage_oca/tests/test_components_base.py]: Fixed!

**Comment by simahawk** [comment]
@jcoux can you squash?
@LoisRForgeFlow could you review this?

**Comment by simahawk** [comment]
I cannot approve my own PR :sweat_smile:

**Comment by simahawk** [comment]
@etobella would you mind to have a look?

**Review by damdam-s — APPROVED** [review]

**Review by SilvioC2C — COMMENTED** [review]
LGTM overall

**Pull request merged**